js/ng.js

```diff
@@ -463,4 +544,92 @@ async function deleteGloss(id=glossHashID) {
         console.log(err)
     })
 
-}
\ No newline at end of file
+}
+
+async function queryBibliographicCitations(glossId) {
+    const query = {
+        "@type": "BibliographicCitation",
+        "references": glossId, 
+        "__rerum.generatedBy": __constants.generator
+    };
+
+    try {
+        const res = await fetch(`${__constants.tiny+"/query"}`, {
+            method: 'POST',
+            mode: 'cors',
+            headers: {
+                "Content-Type": "application/json;charset=utf-8"
+            },
+            body: JSON.stringify(query)
+        }).then(response => response.json());
+
+        return res
+    } catch (err) {
+        console.error(err);
+    }
+}
+
+async function addBibliographicCitationToGloss(bibliographicCitation, glossId) {
+    try {
+        if (typeof bibliographicCitation !== 'string' || !bibliographicCitation.trim()) {
+            const invalidInputEvent = new CustomEvent("Invalid bibliographic citation input: must be a non-empty string.")
+            globalFeedbackBlip(invalidInputEvent, 'Invalid bibliographic citation input: must be a non-empty string.', false)
+            return;
+        }
+
+        const cleanCitation = bibliographicCitation.trim();
+
+        const query = {
+            "@type": "BibliographicCitation",
+            "references": glossId,
+            "citation": cleanCitation, 
+            "__rerum.generatedBy": __constants.generator
+        };
+
+        const existingCitations = await fetch(`${__constants.tiny+"/query"}`, {
+            method: 'POST',
+            mode: 'cors',
+            headers: {
+                "Content-Type": "application/json;charset=utf-8"
+            },
+            body: JSON.stringify(query)
+        }).then(resp => resp.json());
+
+        if (existingCitations.length > 0) {
+            const invalidInputEvent = new CustomEvent("A similar bibliographic citation already exists for this gloss.")
+            globalFeedbackBlip(invalidInputEvent, 'A similar bibliographic citation already exists for this gloss.', false)
+            return;
+        }
+
+        const newCitation = {
+            "@context": "http://www.w3.org/ns/anno.jsonld",
```

Comment:
This does not match the elements `deer-context` value.
`<form id="bibliographicCitationForm" deer-type="BibliographicResource" deer-context="http://purl.org/dc/terms">`